flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ZooKeeperCheckpointIDCounter.java

```diff
@@ -150,32 +171,15 @@ public void setCount(long newId) throws Exception {
 		sharedCount.setCount((int) newId);
 	}
 
-	@VisibleForTesting
-	ConnectionState getLastState() {
-		return connStateListener.lastState;
-	}
-
-	/**
-	 * Connection state listener. In case of {@link ConnectionState#SUSPENDED} or {@link
-	 * ConnectionState#LOST} we are not guaranteed to read a current count from ZooKeeper.
-	 */
-	private static class SharedCountConnectionStateListener implements ConnectionStateListener {
-
-		private volatile ConnectionState lastState;
+	private void checkConnectionState() {
+		final ConnectionState lastState = this.lastState;
```

Comment:
I would not shadow the local variable. One could rename the variable `currentLastState`.